### PR TITLE
Revert breaking change from 625c69e

### PR DESCRIPTION
**Pull request by michielbdejong**

### CODE REVIEW

**Comment by michielbdejong** [comment]
I was [adding `.account`](https://github.com/michielbdejong/ilp-client/blob/master/src/index.js#L384-L385) to work around the bug your PR fixed. :)

File: src/lib/plugin.js [hunk 1]
```diff
@@ -428,20 +428,21 @@ class FiveBellsLedger extends EventEmitter2 {
...
-      throw new errors.InvalidFieldsError('invalid account')
-    }
-    if (message.to && message.account) {
-      throw new errors.InvalidFieldsError('"to" and "account" cannot both be defined')
+    if (typeof message.to !== 'string') {
+      // check for deprecated Message format, from before https://github.com/interledger/rfcs/commit/61958f54c268e5a52e1b85f090df02646b0dda38
+      if (typeof message.account === 'string') {
+        util.deprecate(() => {}, 'switch from using "account" to "to"')()
+        message.to = message.account
+        delete message.account
```
Comment (sharafian) [inline src/lib/plugin.js:444]: This shouldn't modify its parameter

File: src/lib/plugin.js [hunk 2]
```diff
@@ -475,14 +476,15 @@ class FiveBellsLedger extends EventEmitter2 {
...
-      throw new errors.InvalidFieldsError('"to" and "account" cannot both be defined')
-    }
-    if (transfer.account) {
-      util.deprecate(() => {}, 'switch from using "account" to "to"')()
+    if (typeof transfer.to !== 'string') {
+      // check for deprecated Transfer format, from before https://github.com/interledger/rfcs/commit/61958f54c268e5a52e1b85f090df02646b0dda38
+      if (typeof transfer.account === 'string') {
+        util.deprecate(() => {}, 'switch from using "account" to "to"')()
+        transfer.to = transfer.account
+        delete transfer.account
```
Comment (sharafian) [inline src/lib/plugin.js:494]: This shouldn't modify its parameter

**Review by sharafian — CHANGES_REQUESTED** [review]

File: src/lib/plugin.js (hunk 1 above)
Comment (michielbdejong) [reply src/lib/plugin.js:444]: Ah, right. Good point

File: src/lib/plugin.js [hunk 3]
```diff
@@ -428,20 +428,24 @@ class FiveBellsLedger extends EventEmitter2 {
...
-    if (typeof message.to !== 'string' && typeof message.account !== 'string') {
-      throw new errors.InvalidFieldsError('invalid account')
-    }
-    if (message.to && message.account) {
-      throw new errors.InvalidFieldsError('"to" and "account" cannot both be defined')
+    if (typeof message.to !== 'string') {
+      // check for deprecated Message format, from before https://github.com/interledger/rfcs/commit/61958f54c268e5a52e1b85f090df02646b0dda38
+      if (typeof message.account === 'string') {
+        util.deprecate(() => {}, 'switch from using "account" to "to"')()
+        const correctedMessage = Object.assign({}, message)
```
Comment (sharafian) [inline src/lib/plugin.js]: I think it'd be more readable to put that at the top of the function, instead of recursing

File: src/lib/plugin.js [hunk 4]
```diff
@@ -428,20 +428,24 @@ class FiveBellsLedger extends EventEmitter2 {
...
+        correctedMessage.to = message.account
+        delete correctedMessage.account
+        yield * this._sendMessage(correctedMessage)
```
Comment (sharafian) [inline src/lib/plugin.js]: This should just be `yield` instead of `yield  *`, if you want to await `_sendMessage`

**Review by sharafian — CHANGES_REQUESTED** [review]

File: src/lib/plugin.js (hunk 1 above)
Comment (michielbdejong) [reply src/lib/plugin.js:444]: Fixed by line 431 above

File: src/lib/plugin.js (hunk 2 above)
Comment (michielbdejong) [reply src/lib/plugin.js:494]: Fixed by line 485

**Review by sharafian — APPROVED** [review]

**Pull request merged**